FAQ.md

```diff
@@ -0,0 +1,112 @@
+# FAQ
+
+_The first rule of FAQ is don't bikeshed the FAQ, leave that for
+[Create structure for managing docs](https://github.com/golang/dep/issues/331)._
+
+Please contribute to the FAQ! Found an explanation in an issue or pull request helpful?
+Summarize the question and quote the reply, linking back to the original comment.
+
+* [What is a direct or transitive dependency?](#what-is-a-direct-or-transitive-dependency)
+* [Why is it `dep ensure` instead of `dep install`?](#why-is-it-dep-ensure-instead-of-dep-install)
+* [Does `dep` replace `go get`?](#does-dep-replace-go-get)
+* [Why did `dep ensure -update` not update package X?](#why-did-dep-ensure--update-not-update-package-x)
+* [Why is `dep` ignoring the version specified in the manifest?](#why-is-dep-ignoring-the-version-specified-in-the-manifest)
+* [`dep` deleted my files in the vendor directory!](#dep-deleted-my-files-in-the-vendor-directory)
+* [Can I put the manifest and lock in the vendor directory?](#can-i-put-the-manifest-and-lock-in-the-vendor-directory)
+* [Unable to update checked out version: fatal: reference is not a tree](#unable-to-update-checked-out-version-fatal-reference-is-not-a-tree)
+
+## What is a direct or transitive dependency?
+* Direct dependencies are dependencies that are imported by your project.
+* Transitive dependencies are the dependencies of your dependencies. Necessary to compile but are not directly used by your code.
+
+## Why is it `dep ensure` instead of `dep install`?
+
+> Yeah, we went round and round on names. [A lot](https://gist.github.com/jessfraz/315db91b272441f510e81e449f675a8b).
+>
+>  The idea of "ensure" is roughly, "ensure that all my local states - code tree, manifest, lock, and vendor - are in sync with each other." When arguments are passed, it becomes "ensure this argument is satisfied, along with synchronization between all my local states."
+>
+>  We opted for this approach because we came to the conclusion that allowing the tool to perform partial work/exit in intermediate states ended up creating a tool that had more commands, had far more possible valid exit and input states, and was generally full of footguns. In this approach, the user has most of the same ultimate control, but exercises it differently (by modifying the code/manifest and re-running dep ensure).
+-[@sdboyer in #371](https://github.com/golang/dep/issues/371#issuecomment-293246832)
+
+## Does `dep` replace `go get`?
+
```

Comment:
Probably would be good to do a bit more caveating here - this is one of the lesser-explored areas with @rsc so far. So, even though the opinions expressed here are tentative, _only_ having them doesn't really paint a representative picture moving forward.

I'll think about some other verbiage...